stacbuilder/pathparsers.py

```diff
@@ -173,6 +173,62 @@ def type_converters(self) -> TypeConverterMapping:
         return self._type_converters
 
 
+class GeneralInputPathParser(RegexInputPathParser):
```

Comment:
**_Minor remark:_**
I think the name _GeneralInputPathParser_ is a bit misleading. I would propose something like _StandardInputPathParser_.

The name GeneralInputPathParser suggests that this is the general parser that solves _**all**_ cases, and that is not really possible to do. 

In fact, the point of having the pathparsers module is that we we need to allow for a variety of datasets that have different file paths or directory structures, and we can't always control what we will receive. So we need to support that flexibility.

That doesn't mean we cannot have a parser that is good for _most_ cases, as long as those cases follow a common standard. (And that is after all what you are introducing here) So that is my reasoning to propose StandardInputPathParser, or something along those lines.

Maybe a lot of fuss about a word, but when we can squash confusion from the start, then we should.

Reply:
What do you think of DefaultInputParser? As this should be used unless a case is more complex

Reply:
Yes DefaultInputParser is better, it should indeed be the default. Good point.